buffer.c

```diff
@@ -351,3 +351,27 @@ buffer_print(buffer_t *self) {
 
   printf("\n");
 }
+
+/*
+ * Checks if the buffer's end is same than `ends`.
+ */
+
+int
+buffer_ends_w(buffer_t *self, const char *ends) {
+  size_t endslen = strlen(ends);
+  if (endslen > self->len) return -1;
+
+  int advance = self->len - endslen;
+  return (0 == strncmp(self->data + advance, ends, endslen)) ? 1 : -1;
```

Comment:
Why `-1`? 

Wouldn't a boolean value be right for this case?

```suggestion
  return 0 == strncmp(self->data + advance, ends, endslen);
```

Reply:
You're right, I did not do it since I didn't see `<stdbool.h>` anywhere so I wanted to make the code the most similar to the current version as I can.